package.json

```diff
@@ -21,10 +21,10 @@
   },
   "dependencies": {
     "chalk": "^1.1.3",
-    "clone": "^1.0.2",
+    "clone": "^2.1.1",
     "co-fs-extra": "^1.2.1",
     "commander": "^2.6.0",
-    "gray-matter": "^2.1.0",
+    "gray-matter": "^3.1.1",
```

Comment:
Good call.  I’ll check it out.

Reply:
From the [`gray-matter` 3.0 Changelog](https://github.com/jonschlinkert/gray-matter/blob/master/CHANGELOG.md#300---2017-06-30):

> `toml`, `coffee` and `cson` are no longer supported by default. Please see `options.engines` and the examples to learn how to add engines.

Reply:
Hmm.  I don’t think they are widely used, but it’s still a breaking change.  Should I go ahead and revert to `2.x`?

Reply:
Yeah, that's a breaking change. Even though there's no way for users to make those formats the default front matter format, they can still use them by specifying the language in the frontmatter itself: https://github.com/jonschlinkert/gray-matter/tree/2.1.1#optionslang

So either we stick with gray-matter 2 for this release, or we need to re-enable that functionality in gray-matter 3 by manually defining those engines: https://github.com/jonschlinkert/gray-matter#optionsengines

Note that in the next major release of Metalsmith I'm considering just extracting all this functionality into a plugin (#157) so it may not be worth expending too much effort on this now.

Reply:
All of that sounds good to me.  I’ll downgrade the dep, since manually adding back in the engines is likely to have edge cases and problems.

I’m also with you on extracting frontmatter into plugin.  I think the 3.0 docs can include some copy-pasteable commands for some of the most popular plugins (like this one).